specification/cdn/resource-manager/Microsoft.Cdn/stable/2020-09-01/afdx.json

```diff
@@ -0,0 +1,5553 @@
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./cdn.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AFDDomainProperties"
+        }
+      }
+    },
+    "AFDDomainProperties": {
+      "description": "The JSON object that contains the properties of the domain to create.",
+      "required": [
+        "hostName"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/AFDDomainUpdatePropertiesParameters"
+        },
+        {
+          "$ref": "#/definitions/AFDStateProperties"
+        }
+      ],
+      "properties": {
+        "domainValidationState": {
+          "description": "Provisioning substate shows the progress of custom HTTPS enabling/disabling process step by step. DCV stands for DomainControlValidation.",
+          "readOnly": true,
+          "enum": [
+            "Unknown",
+            "Submitting",
+            "Pending",
+            "TimedOut",
+            "PendingRevalidation",
+            "Approved"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DomainValidationState",
+            "modelAsString": true
+          }
+        },
+        "hostName": {
+          "description": "The host name of the domain. Must be a domain name.",
+          "type": "string"
+        },
+        "validationProperties": {
+          "description": "Values the customer needs to validate domain ownership",
+          "readOnly": true,
+          "type": "object",
+          "$ref": "#/definitions/DomainValidationProperties"
+        }
+      }
+    },
+    "AFDDomainUpdateParameters": {
+      "description": "The domain JSON object required for domain creation or update.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AFDDomainUpdatePropertiesParameters"
+        }
+      }
+    },
+    "AFDDomainUpdatePropertiesParameters": {
+      "description": "The JSON object that contains the properties of the domain to create.",
+      "properties": {
+        "tlsSettings": {
+          "description": "The configuration specifying how to enable HTTPS for the domain - using AFDX managed certificate or user's own certificate. If not specified, enabling ssl uses AFDX managed certificate by default.",
+          "type": "object",
+          "$ref": "#/definitions/AFDDomainHttpsParameters"
+        },
+        "azureDnsZone": {
+          "description": "Resource reference to the Azure DNS zone",
+          "type": "object",
+          "$ref": "./cdn.json#/definitions/ResourceReference"
+        }
+      }
+    },
+    "DomainValidationProperties": {
+      "description": "The JSON object that contains the properties to validate a domain.",
+      "properties": {
+        "validationToken": {
```

Comment:
Is this a secret? Then please annotate with x-ms-secret.

Reply:
The response is not a secret.